test/versioned/langchain/vectorstore.tap.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper')
+const createOpenAIMockServer = require('../openai/mock-server')
+
+const config = {
+  ai_monitoring: {
+    enabled: true
+  },
+  feature_flag: {
+    langchain_instrumentation: true
+  }
+}
+
+tap.test('Langchain instrumentation - vectorstore', (t) => {
+  t.autoend()
+
+  t.beforeEach(async (t) => {
+    const { host, port, server } = await createOpenAIMockServer()
+    t.context.server = server
+    t.context.agent = helper.instrumentMockedAgent(config)
+    const { OpenAIEmbeddings } = require('@langchain/openai')
+
+    t.context.embedding = new OpenAIEmbeddings({
+      openAIApiKey: 'fake-key',
+      configuration: {
+        baseURL: `http://${host}:${port}`
+      }
+    })
+  })
+
+  t.afterEach(async (t) => {
+    t.context?.server?.close()
+    helper.unloadAgent(t.context.agent)
+    // bust the require-cache so it can re-instrument
+    Object.keys(require.cache).forEach((key) => {
+      if (key.includes('@langchain/core') || key.includes('openai')) {
+        delete require.cache[key]
+      }
+    })
+  })
+
+  t.test('should create vectorstore events for every similarity search call', (t) => {
+    const { agent, embedding } = t.context
+    const { MemoryVectorStore } = require('langchain/vectorstores/memory')
```

Comment:
they are trying to move towards using `@langchain/community`, you might want to use vector stores from there.  For some reason this one isn't in there though

Reply:
Since `MemoryVectorStore` is not in `@langchain/community`, I can pick another vector store that's in `@langchain/community` in these tests.

Reply:
Closing comment. Swapped for elastic search.